### DISCUSSION TITLE
`vmap` with ragged data (list of variable length arrays)

### DISCUSSION THREAD

**Discussion by 8bitmp3** [jax-ml/jax#5184] [Q&A]
Bumped into a question by @iskandr (I think)
> how do use `vmap` with ragged data (list of variable length arrays)?

Thought I'd share it here and try to get an answer to build up the "knowledge base" under `github/jax` 🤔

Source: https://twitter.com/iskander/status/1296906143332261889

**Comment by charlessievers** [comment]
Not an answer, but I would like to know if there are any ways to vmap and handle lists of variable length arrays.

**Comment by jakevdp** [comment]
JAX does not currently have any facility for computation on ragged arrays, and the `vmap` transform in particular is built with homogeneous batches in mind. Depending on your use case, you may be able to do what you have in mind using JAX's [experimental sparse array support](https://jax.readthedocs.io/en/latest/jax.experimental.sparse.html); but although similar in spirit, sparse arrays are not precisely equivalent to ragged arrays.

**Reply by charlessievers** [reply]
Thanks for the speedy reply! I will give it a look.

**Comment by cevheck** [comment]
@jakevdp,
Any updates on this issue? I would like to perform a certain calculation on different machine learning models (ensemble method) and do them together in a single vmap call. My issue is that each of those models has a different length of parameters (e.g. combining a single layer 16neurons network with a double layer 32 neurons network). 

Thanks in advance!

**Reply by jakevdp** [reply]
There's been more work on ragged arrays recently (e.g. #16072 and #16324), but it's still all very experimental. @axch and @mattjj know more.
